pkg/eigenState/stakerShares/stakerShares.go

```diff
@@ -26,27 +26,29 @@ import (
 	"gorm.io/gorm/clause"
 )
 
-type StakerShares struct {
-	Staker      string
-	Strategy    string
-	Shares      string
-	BlockNumber uint64
-	CreatedAt   time.Time
+var WAD = big.NewInt(1e18)
+
+type StateDiffEvent interface {
+	SlotID() types.SlotID
 }
 
-type AccumulatedStateChange struct {
-	Staker      string
-	Strategy    string
-	Shares      *big.Int
-	BlockNumber uint64
+type StateDiff struct {
+	Event StateDiffEvent
```

Comment:
As I got further down, I realized what you're doing, and this isnt good lol.

While this makes sense in a polymorphic language, this is not a good pattern for Go. You should be able to use generics and do something like:

```
type StateDiff[T any] struct {
    Event T
}
```
but even that isn't friendly....

The problem here is that since in Go you cant describe the contents of the struct you're trying to generically represent, you're basically treating `StateDiffEvent` as `interface{}` which isnt clear at all.

Reply:
https://github.com/Layr-Labs/sidecar/pull/68/commits/849bd0db77bd553611d6ccdf776de875882dc9c5